public/app/features/dashboard/state/DashboardMigrator.ts

```diff
@@ -67,6 +67,34 @@ export class DashboardMigrator {
     this.dashboard = dashboardModel;
   }
 
+  /**
+   * When changing default datasource which is stored as null Grafana get's into a mixed state where queries have
+   * data source uid & type set that is different from the now new default
+   */
+  syncQueryDataSources() {
+    for (const panel of this.dashboard.panels) {
+      // only interested in panels that use default (null) data source
+      if (panel.datasource) {
+        continue;
+      }
+
+      const defaultDS = getDataSourceSrv().getInstanceSettings(null);
+      // if default ds is mixed then skip this
+      if (!defaultDS || defaultDS.meta.mixed) {
+        return;
+      }
```

Comment:
this could be outside the panel loop.

:thinking: but `panel.datasource` should be set most of the time, so maybe this is better.